PEM - Update pgbouncer as per DF-319 (#3513)
EnterpriseDB/docs: product_docs/docs/pem/9/considerations/pem_pgbouncer/configuring_pgBouncer.mdx

    - /pem/latest/pem_pgbouncer/03_configuring_pgBouncer/
---

> You must configure PgBouncer to work with the PEM database server. The name and location of directories and files depend on the repository used to install PgBouncer. You can install PgBouncer using the PostgreSQL community or the EDB repo.

Reviewer: ```suggestion
You must configure PgBouncer to work with the PEM database server. 

The name and location of the directories and files in the configuration steps depend on whether you installed the community version of PgBouncer or EDB PgBouncer. The instructions assume you are using EDB PgBouncer. If you have installed PgBouncer from the community repo, replace the names of the files and directories in the example with the PgBouncer values.
```

> Here is the list of names and locations of directories and files:

Reviewer: You can remove this.
```suggestion

```

| Name                | PgBouncer                          | EDB PgBouncer                          |
|---------------------|------------------------------------|----------------------------------------|
| Repo                | PostgreSQL community repo          | EDB repo                               |

> | PgBouncer directory | `/etc/pgbouncer<x.x>`              | `/etc/edb/pgbouncer<x.>`               |

Reviewer: ```suggestion
| PgBouncer directory | `/etc/pgbouncer<x.x>`              | `/etc/edb/pgbouncer<x.x>`               |
```

    - /pem/latest/pem_pgbouncer/03_configuring_pgBouncer/
---

> You must configure PgBouncer to work with the PEM database server. This example runs PgBouncer as the enterprisedb system user and outlines the process of configuring pgBouncer version `<x.x>` or later.

Reviewer: ```suggestion
You must configure PgBouncer to work with the PEM database server. This example runs PgBouncer as the enterprisedb system user and outlines the process of configuring pgBouncer.
```
Saying version x.x or later isn't adding any value. I'd just remove it.

> 1.  Open a terminal window and navigate to the pgBouncer directory.

Reviewer: ```suggestion
1.  Open a terminal window and navigate to the PgBouncer directory.
```
Is the etc directory a subdirectory to the "PgBouncer" directory? If so, it might be clearer to saying "PgBouncer installation directory" above.

Author: we mean /etc/edb/pgbouncer as pgbouncer directory

> 1.  Open a terminal window and navigate to the PgBouncer directory.

Reviewer: ```suggestion
1.  Open a terminal window and navigate to the PgBouncer directory:
    - If you've installed the community version of PgBouncer using the PostgreSQL repo, the directory is `/etc/pgbouncer<x.x>.
    -  If you've installed EDB PgBouncer using the EDB repo, the directory is `/etc/edb/pgbouncer<x.x>. This example assumes you installed EDB PgBouncer.
```

2.  Change the owner of the `etc` directory for pgBouncer (where `pgbouncer.ini` resides) to `enterprisedb`, and change the directory permissions to `0700`:

    ```shell
    $ chown enterprisedb:enterprisedb /etc/edb/pgbouncer<x.x>
    $ chmod 0700 /etc/edb/pgbouncer<x.x>
    ```

> !!! Note

Reviewer: See my suggestion for an alternative approach above.

>     If you install pgbouncer using the PostgreSQL repo, the pgBouncer directory is `/etc/pgbouncer<x.x>`.

Reviewer: ```suggestion
    If you've installed the community version of PgBouncer using the PostgreSQL repo instead of installing EDB PgBouncer from the EDB repo, the PgBouncer directory is `/etc/pgbouncer<x.x>`.
```

> 3.  Change the contents of the `pgbouncer.ini` or `edb-pgbouncer.ini` file:

Reviewer: ```suggestion
3.  Change the contents of the `edb-pgbouncer.ini` file:
```
The changes we describe apply to the edb-pgbouncer.ini file. I think it is a little confusing to show the changes for that file when we mention the other one too. We could add more content around the differences between in the config for the two versions of PgBouncer (paths and filenames), probably up front, or it is probably easier for the reader and the author to have two sets of instructions.

    server_idle_timeout = 60
    ```

!!! Note

>     For more information on `auth_user` see [authentication settings](https://www.pgbouncer.org/config.html#authentication-settings).

Reviewer: ```suggestion
    For more information on `auth_user` see [Authentication settings](https://www.pgbouncer.org/config.html#authentication-settings).
```

4.  Create and update the `/etc/edb/pgbouncer<x.x>/userlist.txt` authentication file for PgBouncer:

    ```sql
    ## Connect to pem database as a superuser, 
    ## create the userslist.txt file and add 
    ## username and their password list in CSV format
    $ psql -p 5444 -d pem -U enterprisedb
    
    pem=# COPY (
    SELECT 'pgbouncer'::TEXT, 'pgbouncer_password'
    UNION ALL
    SELECT 'pem_admin1'::TEXT, 'pem_admin1_password'
    TO '/etc/edb/pgbouncer<x.x>/userlist.txt'
    WITH (FORMAT CSV, DELIMITER ' ', FORCE_QUOTE *);
    __OUTPUT__
    COPY 2
    ```

    This creates `/etc/edb/pgbouncer<x.x>/userlist.txt` file and adds the username and password list.

    !!! Note

>         A superuser can't invoke the PEM authentication query function `pem.get_proxy_auth(text)`. If the pem_admin user is a superuser, you must add the password to the authentication file (`enterprisedb` in the example).

Reviewer: It isn't completely clear how these two sentences relate. I think what we are trying to say is:
```suggestion
        If the pem_admin user is a superuser, you must add the password to the authentication file (`enterprisedb` in the example). This allows the pem_admin user to invoke the PEM authentication query function `pem.get_proxy_auth(text)`.
```
